### ISSUE TITLE
Incorrect unused variable error

### ISSUE THREAD

**Issue by optical** [palantir/tslint#325]
Very similar, if not the same as #237

Using the following contrived typescript file

```ts
    /// <reference path="typings/node/node.d.ts" />
    import fs = require('fs');
    import createWriteStream = fs.createWriteStream;
    console.log(createWriteStream);
```

With these rules

```json
    {
      "rules": {
        "no-unused-variable": true
      }
    }
```

We get this error about fs being unused:
`test.ts[1, 8]: unused variable: 'fs`

This has been the behavior since 1.0

**Comment by nkzawa** [comment]
Maybe it's the same problem, but the following case incorrectly becomes an error as well.

```ts
import foo = require("./foo");
export import bar = foo.bar;
```

**Comment by ashwinr** [comment]
looks like a problem with the compiler's language services. i'll file a bug on them.

**Comment by vsiao** [comment]
@ashwinr, can you link to the bug that you filed?

**Comment by falsandtru** [comment]
I also the same problem occurs. Variable has been used. I want fix to it.

**Comment by leeavital** [comment]
This is in fact a problem with typescript services. Here's the issue on Microsoft/TypeScript.

https://github.com/Microsoft/TypeScript/issues/3688

**Comment by adidahiya** [comment]
Looks like the TS issue has been fixed upstream on their release-1.5 branch. Will be resolved when #508 merges.

**Comment by adidahiya** [comment]
Nevermind, this is not fixed yet.

Are there real use cases for this besides the contrived examples above? Why would you want to use an imported value in another `import` statement?

**Comment by optical** [comment]
We encountered this in some of our projects where we wanted to import Classes etc straight out of the module without needing to refer to them via `module.Class`. Might not be as much of an issue now if you use the ES6 import syntax.

**Comment by Necroskillz** [comment]
fyi this is also an `no-unused-variable` error, and shouldn't be. Not sure if it's related, but there are some issues about this rule and none of them mentions it.

```
import * as cache from './utils/cache';

export { cache };
```
```
unused variable: 'cache'
```

**Comment by JKillian** [comment]
Related to #570

**Comment by praveennet** [comment]
@adidahiya just to capture a usecase that we are having in this issue. As mentioned in #651 - one of use case for us is that we are using internal frameworks which have deep nested classes. code looks too long if we had to use full qualified name everywhere.

e.g. this looks better
```typescript
import ExternalDefinition = require("ExternalDefinition");
import Def = ExternalDefinition.MyNamespace.MyClass1;

...
var myobj: Def.Interface;
Def.MyMethod();

...
}
```

e.g. this looks **not** so better
```typescript
import ExternalDefinition = require("ExternalDefinition");

...
var myobj: ExternalDefinition.MyNamespace.MyClass1.Interface;
ExternalDefinition.MyNamespace.MyClass1.MyMethod();
...
```

**Comment by adidahiya** [comment]
@praveennet yep, those are valid use cases. Can you check if this is still a bug in the latest TSLint?

**Comment by praveennet** [comment]
I verified that with v2.5.0 this issue still repros.

**Comment by JKillian** [comment]
@praveennet It's definitely a bug, but the problem seems to be a TypeScript one at the moment (see https://github.com/Microsoft/TypeScript/issues/3688#issuecomment-140119518). Unfortunately nothing we can do to fix it on our end for now

**Comment by HamletDRC** [comment]
So are you just waiting on TypeScript to fix the problem? 
We fixed this problem ourselves in our unused-import rule: https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/noUnusedImportsRule.ts

Any chance of a temporary workaround in the tslint rule?

**Comment by davidreher** [comment]
Hey I am getting the same error on the following code:
```javascript
import foo from './bar';
export {foo}
```
At least if no other usages except for the export are present ... What works is
```
export {default as foo} from './bar';
```
Is this the same issue or is it another issue or is it no issue at all?

**Comment by adidahiya** [comment]
@davidreher that looks like #684. Using `typescript@next` alongside TSLint should fix the problem.

**Comment by davidreher** [comment]
@adidahiya thx for the help, that is indeed my issue, and your fix worked ;) Hope the final 1.8 gets out soon ...
